### PR TITLE
Fix bug #1942 Rename/split Bag to Cells 

### PR DESCRIPTION
**Pull request by chmx0929**
To distinguish data structure "Bag" and the syntax "Bag". We rename the sort name "Bag", defined in module KCELLS in kast.k, to CellBag. Then, we fix corresponding errors. For all tests that use module KCELLS in kast.k, especially tests in RuleGrammarTest.java, are perfect tests to check whether the program still works after we do this change.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/148313234996210/224288563469573)

### CODE REVIEW

File: kernel/src/main/java/org/kframework/kore/compile/checks/CheckSortTopUniqueness.java [hunk 1]
```diff
@@ -52,10 +52,11 @@ private void check(Production p) {
     }
 
     private void check(Sort s, Sentence p) {
+        //Term Bag here stands for syntax Bag, so we rename it to CellBag to distinguish data structure Bag and syntax Bag
```
Comment (radumereuta) [inline kernel/src/main/java/org/kframework/kore/compile/checks/CheckSortTopUniqueness.java]: Comments in the file should reflect what is in the file. This comment should be in something like a changelog.

File: kernel/src/main/java/org/kframework/parser/concrete2kore/generator/RuleGrammarGenerator.java [hunk 2]
```diff
@@ -310,7 +310,8 @@ public Module processHybridModule(Module m) {
                                 if (cfgInfo.isLeafCell(p.sort())) {
                                     body = p.items().tail().head();
                                 } else {
-                                    body = NonTerminal(Sort("Bag"));
+                                    //Term Bag here stands for syntax Bag, so we rename it to CellBag to distinguish data structure Bag and syntax Bag
```
Comment (radumereuta) [inline kernel/src/main/java/org/kframework/parser/concrete2kore/generator/RuleGrammarGenerator.java]: Same as above.

File: kernel/src/test/java/org/kframework/parser/concrete2kore/RuleGrammarTest.java [hunk 3]
```diff
@@ -35,6 +35,8 @@
     private final static Sort startSymbol = DefinitionParsing.START_SYMBOL;
     private String baseKText;
 
+    // Tests in this file test rule grammar. And they are perfect tests to test whether the program works
+    // if we rename syntax Bag to CellBag.
```
Comment (radumereuta) [inline kernel/src/test/java/org/kframework/parser/concrete2kore/RuleGrammarTest.java]: Same as above.

**Review by radumereuta — CHANGES_REQUESTED** [review]
This change actually impacts every definition we have in K right now, since this file is imported and used in definitions to parse rules and configuration cells. So all tests should pass. I don't know why codeship didn't start testing this PR.

**Comment by HelenRouty** [comment]
Thank you. Where should we create a change log? Is the change log pull request?

**Comment by HelenRouty** [comment]
## Change Log
We changed 4 "Bag" to "CellBag" in:
(1) check method in _kernel/src/main/java/org/kframework/kore/compile/checks/CheckSortTopUniqueness.java_
(2) processHybridModule method in _kernel/src/main/java/org/kframework/parser/concrete2kore/generator/RuleGrammarGenerator.java_
(3) syntax in _k-distribution/include/builtin/kast.k_
(4) test cases in _kernel/src/test/java/org/kframework/parser/concrete2kore/RuleGrammarTest.java_

## Tests
Rule grammer tests in the file _k/kernel/src/test/java/org/kframework/parser/concrete2kore/RuleGrammarTest.java_ are perfect tests to test whether the program works if we rename syntax Bag to CellBag.

**Comment by cos** [comment]
@radumereuta, could you have a quick look at this?

**Comment by radumereuta** [comment]
The changes look right, but I don't see the tests passing, or even starting.
@cos do you know anything about that?

**Comment by daejunpark** [comment]
Close this PR exercise in the class.

**Pull request closed without merging**